components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6008,14 +6014,33 @@ where
             RegionChangeEvent::UpdateBuckets(buckets_count),
             self.fsm.peer.get_role(),
         );
-        let old_region_buckets = self.fsm.peer.region_buckets.replace(region_buckets);
+        let old_region_buckets: Option<BucketStat> =
+            self.fsm.peer.region_buckets.replace(region_buckets);
         self.fsm.peer.last_region_buckets = old_region_buckets;
         let mut store_meta = self.ctx.store_meta.lock().unwrap();
+        let version = self.fsm.peer.region_buckets.as_ref().unwrap().meta.version;
         if let Some(reader) = store_meta.readers.get_mut(&self.fsm.region_id()) {
             reader.update(ReadProgress::region_buckets(
                 self.fsm.peer.region_buckets.as_ref().unwrap().meta.clone(),
             ));
         }
+
+        // Notify followers to refresh their buckets version
+        let peers = self.region().get_peers().to_vec();
+        for p in peers {
+            if &p == self.peer() || p.is_witness {
```

Comment:
Should it check leadership first before sending messages?

Reply:
yes， I will add this checker